src/main/java/org/jenkinsci/plugins/workflow/flow/DurabilityHintProvider.java

```diff
@@ -0,0 +1,36 @@
+package org.jenkinsci.plugins.workflow.flow;
+
+import hudson.ExtensionList;
+import hudson.ExtensionPoint;
+import hudson.model.Item;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+
+/**
+ * Provides a way to indirectly register durability settings to apply to pipelines.
+ */
+public interface DurabilityHintProvider extends ExtensionPoint {
+
+
+    public int ordinal();
+
+    @CheckForNull
+    FlowDurabilityHint suggestFor(@Nonnull Item x);
+
+    public static @Nonnull FlowDurabilityHint suggestedFor(@Nonnull Item x) {
+        int ordinal = Integer.MAX_VALUE;
+        FlowDurabilityHint hint = GlobalDefaultFlowDurabilityLevel.getDefaultDurabilityHint();
```

Comment:
`GlobalDefaultFlowDurabilityLevel` could be made to have a `DurabilityHintProvider` for consistency.

Reply:
Implementing it this way allows delegating to GlobalDefaultFlowDurabilityLevel if other implementations do not provide a value - no real benefit to changing it here.